modules/tinymce/src/plugins/table/main/ts/ui/UiUtils.ts

```diff
@@ -24,7 +28,10 @@ const onSetupToggle = (editor: Editor, formatName: string, formatValue: string)
       const setActive = (matched: boolean) =>
         api.setActive(isNone ? !matched : matched);
 
-      setActive(editor.formatter.match(formatName, { value: formatValue }, undefined, isNone));
+      const selectedCells = TableSelection.getCellsFromSelection(Util.getSelectionStart(editor), selections, Util.getIsRoot(editor));
+      const forAll = () => Arr.forall(selectedCells, (cell) => editor.formatter.match(formatName, { value: formatValue }, cell.dom));
+      const forAny = () => Arr.exists(selectedCells, (cell) => editor.formatter.match(formatName, { value: formatValue }, cell.dom, isNone));
+      setActive(isNone ? forAny() : forAll());
```

Comment:
Maybe breaking it up into an if statement could simplify things?
```suggestion
      if (isNone) {
        setActive(Arr.exists(selectedCells, (cell) => editor.formatter.match(formatName, { value: formatValue }, cell.dom, true)));
      } else {
        setActive(Arr.forall(selectedCells, (cell) => editor.formatter.match(formatName, { value: formatValue }, cell.dom)));
      }
```

Reply:
This also should be using `matchNode` not `match` as we only want to look at this specific node and not it's parent nodes.